From abb9efc70988087a7ea04c90112657e68e8894a8 Mon Sep 17 00:00:00 2001
Message-Id: <abb9efc70988087a7ea04c90112657e68e8894a8.1640138242.git.geliang.tang@suse.com>
From: Gal Hammer <ghammer@redhat.com>
Date: Wed, 6 Jan 2021 11:28:26 +0200
Subject: [PATCH] igb: fix TDBAL register show incorrect value
Git-commit: abb9efc70988087a7ea04c90112657e68e8894a8
Patch-mainline: v5.12-rc2
References: bsn#47


Fixed a typo which caused the registers dump function to read the
RDBAL register when printing TDBAL register values.

Signed-off-by: Gal Hammer <ghammer@redhat.com>
Tested-by: David Switzer <david.switzer@intel.com>
Signed-off-by: Tony Nguyen <anthony.l.nguyen@intel.com>
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 drivers/net/ethernet/intel/igb/igb_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/intel/igb/igb_main.c b/drivers/net/ethernet/intel/igb/igb_main.c
index 84d4284b8b32..d6090faaa41d 100644
--- a/drivers/net/ethernet/intel/igb/igb_main.c
+++ b/drivers/net/ethernet/intel/igb/igb_main.c
@@ -316,7 +316,7 @@ static void igb_regdump(struct e1000_hw *hw, struct igb_reg_info *reginfo)
 		break;
 	case E1000_TDBAL(0):
 		for (n = 0; n < 4; n++)
-			regs[n] = rd32(E1000_RDBAL(n));
+			regs[n] = rd32(E1000_TDBAL(n));
 		break;
 	case E1000_TDBAH(0):
 		for (n = 0; n < 4; n++)
-- 
2.31.1

